common/remote_object.go

```diff
@@ -227,42 +227,48 @@ func valueFromRemoteObject(_ context.Context, robj *cdpruntime.RemoteObject) (an
 	return val, err
 }
 
-func parseConsoleRemoteObjectPreview(logger *log.Logger, op *cdpruntime.ObjectPreview) string {
+func parseConsoleRemoteObjectPreview(logger *log.Logger, op *cdpruntime.ObjectPreview) (string, error) {
 	obj := make(map[string]string)
 	if op.Overflow {
 		logger.Infof("parseConsoleRemoteObjectPreview", "object is too large and will be parsed partially")
 	}
 
 	for _, p := range op.Properties {
-		val := parseConsoleRemoteObjectValue(logger, p.Type, p.Subtype, p.Value, p.ValuePreview)
+		val, err := parseConsoleRemoteObjectValue(logger, p.Type, p.Subtype, p.Value, p.ValuePreview)
+		if err != nil {
+			return "", err
+		}
 		obj[p.Name] = val
 	}
 
 	bb, err := json.Marshal(obj)
 	if err != nil {
-		logger.Errorf("parseConsoleRemoteObjectPreview", "failed to marshal object to string: %v", err)
+		return "", fmt.Errorf("marshaling object to string: %w", err)
```

Comment:
Here is one more marshaling error that we might want to enrich with the marshaled object.

Reply:
Changed in b0f41e70076a21a5bb9a2007e5b213785a80bd95